web/src/components/snapshots/SnapshotStorageDestination.tsx

```diff
@@ -496,12 +576,16 @@ class SnapshotStorageDestination extends Component<Props, State> {
     field: FieldName,
     e: React.ChangeEvent<HTMLInputElement>
   ) => {
-    let nextState: State = {};
+    let nextState: {
+      [K in FieldName]?: string | boolean;
+    } = {};
     if (field === "useIamAws" || field === "gcsUseIam") {
       nextState[field] = e.target.checked;
     } else {
       nextState[field] = e.target.value;
     }
+    // TODO: make this more explicit
+    // @ts-ignore
```

Comment:
I don't know enough about `Pick` to get this to work yet. Will return to this later.